# HG changeset patch
# User Thomas D. Dean <tomdean@speakeasy.org>
# Date 1233859349 18000
#      Thu Feb 05 13:42:29 2009 -0500
# Node ID 6d15bc6c4c15f813cae0650d14750ebedc1d99f8
# Parent  e4f89f4a7cf88d50fcb32635404c11e56e35a30d
polyout.m: Replace com2str with num2str.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-02-05  Thomas D. Dean  <tomdean@speakeasy.org>
+
+	* polynomial/polyout.m: Replace com2str with num2str, and minor
+	style changes.
+
 2009-02-04  John W. Eaton  <jwe@octave.org>
 
 	* help/which.m: Still print something sensible if type is empty.
 
 2009-02-04  Soren Hauberg  <hauberg@gmail.com>
             Thomas Treichl  <Thomas.Treichl@gmx.net>
 
 	* miscellaneous/Makefile.in (SOURCES): Add bzip2.m to the list.
diff --git a/scripts/polynomial/polyout.m b/scripts/polynomial/polyout.m
--- a/scripts/polynomial/polyout.m
+++ b/scripts/polynomial/polyout.m
@@ -48,25 +48,24 @@ function y = polyout (c, x)
   endif
 
   if (! isvector (c))
     error("polyout: first argument must be a vector");
   endif
 
   if (nargin == 1)
     x = "s";
-  elseif (! ischar(x))
+  elseif (! ischar (x))
     error("polyout: second argument must be a string");
   endif
 
-  n = length(c);
+  n = length (c);
   if(n > 0)
     n1 = n+1;
 
-
     tmp = coeff (c(1));
     for ii = 2:n
       if (real (c(ii)) < 0)
 	ns = " - ";
 	c(ii) = -c(ii);
       else
         ns = " + ";
       endif
@@ -74,26 +73,26 @@ function y = polyout (c, x)
       tmp = sprintf ("%s*%s^%d%s%s", tmp, x, n1-ii, ns, coeff (c(ii)));
 
     endfor
   else
     tmp = " ";
   endif
 
   if(nargout == 0)
-    disp(tmp)
+    disp (tmp)
   else
     y = tmp;
   endif
 
 endfunction
 
-function str = coeff(c)
+function str = coeff (c)
   if (imag (c))
     if (real (c))
-      str = sprintf ("(%s)", com2str(c));
+      str = sprintf ("(%s)", num2str (c, 5));
     else
-      str = com2str(c);
+      str = num2str (c, 5);
     endif
   else
-    str = num2str(c);
+    str = num2str (c, 5);
   endif
-
+endfunction
